diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/PackageNaming.kt

```diff
@@ -218,8 +218,14 @@ class PackageNaming(configRules: List<RulesConfig>) : DiktatRule(
             ?: run {
                 // there is missing package statement in a file, so it will be created and inserted
                 val newPackageDirective = generatedPackageDirective.findLeafWithSpecificType(PACKAGE_DIRECTIVE)!!
-                packageDirectiveNode.treeParent.replaceChild(packageDirectiveNode, newPackageDirective)
-                newPackageDirective.treeParent.addChild(PsiWhiteSpaceImpl("\n\n"), newPackageDirective.treeNext)
+                val packageDirectiveParent = packageDirectiveNode.treeParent
+                packageDirectiveParent.replaceChild(packageDirectiveNode, newPackageDirective)
+                val insertWhitespaceBefore = if (newPackageDirective != packageDirectiveParent.firstChildNode) {
```

Comment:
Maybe, check whether there is a white space node before and after package directive node? Actually, these `\n\n` are not responsibility of this rule, so here we should ensure that the tree is correct, and FileStructureRule will do the rest.

Reply:
But shouldn't each rule work independently? What if user disable `FileStructureRule` ?

Reply:
Yes, rules should work independently. My main suggestion was to check, whether there are white spaces both before and after the package directive node; I think we can make two newlines on both sides.

Reply:
Got it. Yes, I agree